jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/TopicDraftHibernateDaoTest.java

```diff
@@ -0,0 +1,118 @@
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.model.dao.hibernate;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.classic.Session;
+import org.jtalks.jcommune.model.dao.TopicDraftDao;
+import org.jtalks.jcommune.model.entity.JCUser;
+import org.jtalks.jcommune.model.entity.ObjectsFactory;
+import org.jtalks.jcommune.model.entity.PersistedObjectsFactory;
+import org.jtalks.jcommune.model.entity.TopicDraft;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTransactionalTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionConfiguration;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+/**
+ * @author Dmitry S. Dolzhenko
+ */
+@ContextConfiguration(locations = {"classpath:/org/jtalks/jcommune/model/entity/applicationContext-dao.xml"})
+@TransactionConfiguration(transactionManager = "transactionManager", defaultRollback = true)
+@Transactional
+public class TopicDraftHibernateDaoTest extends AbstractTransactionalTestNGSpringContextTests {
+    @Autowired
+    private SessionFactory sessionFactory;
+    @Autowired
+    private TopicDraftDao dao;
+
+    private Session session;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        session = sessionFactory.getCurrentSession();
+        PersistedObjectsFactory.setSession(session);
+    }
+
+    /*===== Common methods =====*/
+
+    @Test
+    public void testGet() {
+        TopicDraft topicDraft = PersistedObjectsFactory.getDefaultTopicDraft();
+
+        TopicDraft result = dao.get(topicDraft.getId());
+
+        assertNotNull(result);
+        assertEquals(result.getId(), topicDraft.getId());
+    }
+
+
+    @Test
+    public void testGetInvalidId() {
+        TopicDraft result = dao.get(-567890L);
+
+        assertNull(result);
+    }
+
+    @Test
+    public void testUpdate() {
+        final String newTitle = "new title";
+        final String newContent = "new content";
+
+        TopicDraft topicDraft = PersistedObjectsFactory.getDefaultTopicDraft();
+
+        topicDraft.setTitle(newTitle);
+        topicDraft.setContent(newContent);
+        dao.saveOrUpdate(topicDraft);
+
+        session.flush();
+        session.evict(topicDraft);
+
+        TopicDraft result = (TopicDraft) session.get(TopicDraft.class, topicDraft.getId());
+
+        assertEquals(result.getTitle(), newTitle);
+        assertEquals(result.getContent(), newContent);
+    }
+
+    @Test(expectedExceptions = Exception.class)
```

Comment:
Should be a concrete exception. The test mail fail with NPE - we don't wan to expect it :)